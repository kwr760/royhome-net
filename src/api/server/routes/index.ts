import getResumeHandler from '../handler/resume/get-resume';
import putLogHandler from '../handler/log/put-log';
import { RouteType } from '../../types/routes.types';

const routes: RouteType[] = [
  {
    method: 'get',
    path: '/resume/:email',
    handler: getResumeHandler,
    fetchData: {
      resume: getResumeHandler,
    },
  },
  {
    method: 'put',
    path: '/log',
    handler: putLogHandler,
  },
];

export default routes;
