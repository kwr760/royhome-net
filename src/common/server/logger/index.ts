import LOG_LEVELS from '../../util/logger/logger-levels';
import log from './log';

class Logger {
  writeLog = (msg) => {
    log(msg);
  };

  debug = (msg) => {
    this.writeLog({
      logType: LOG_LEVELS.DEBUG,
      msg,
    });
  };

  log = (msg) => {
    this.writeLog({
      logType: LOG_LEVELS.INFO,
      msg,
    });
  };

  warning = (msg) => {
    this.writeLog({
      logType: LOG_LEVELS.WARN,
      msg,
    });
  };

  error = (msg) => {
    this.writeLog({
      logType: LOG_LEVELS.ERROR,
      msg,
    });
  };

  fatal = (msg) => {
    this.writeLog({
      logType: LOG_LEVELS.FATAL,
      msg,
    });
  };
}

export default new Logger();
