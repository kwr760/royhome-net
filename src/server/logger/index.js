import LOG_LEVELS from '../../logger/levels';
import log from './log';

class Logger {
  debug = (msg) => {
    log({
      level: LOG_LEVELS.DEBUG,
      msg,
    });
  };

  log = (msg) => {
    log({
      level: LOG_LEVELS.INFO,
      msg,
    });
  };

  warning = (msg) => {
    log({
      level: LOG_LEVELS.WARN,
      msg,
    });
  };

  error = (msg) => {
    log({
      level: LOG_LEVELS.ERROR,
      msg,
    });
  };

  fatal = (msg) => {
    log({
      level: LOG_LEVELS.FATAL,
      msg,
    });
  };
}

export default new Logger();
