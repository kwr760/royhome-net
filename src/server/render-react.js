import path from 'path';
import fs from 'fs';

import { INTERNAL_SERVER_ERROR } from 'http-status-codes';

import React from 'react';
import ReactDOMServer from 'react-dom/server';
import { StaticRouter, Route } from 'react-router-dom';

import App from '../web/components/App';
import Logger from './logger';

const renderReact = (req, res) => {
  const context = {};
  const appRenderer = ReactDOMServer.renderToString(
    <StaticRouter location={req.url} context={context}>
      <Route component={App} />
    </StaticRouter>,
  );
  Logger.log(req.url);
  Logger.log(appRenderer);
  Logger.log(context);

  const indexFile = path.resolve('./dist/public/index.html');
  fs.readFile(indexFile, 'utf8', (err, data) => {
    if (err) {
      console.error(`The following error: ${err}`);
      return res.sendStatus(INTERNAL_SERVER_ERROR);
    }

    return res.send(
      data.replace(
        '<div id="root"></div>',
        `<div id="root">${appRenderer}</div>`,
      ),
    );
  });
};

export default renderReact;
