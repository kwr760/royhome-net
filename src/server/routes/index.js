import getResumeHandler from './resume/get';
import getPrivateHandler from './private/get';
import getCoursesHandler from './courses/get';
import getAdminHandler from './admin/get';

const routes = [
  {
    method: 'get',
    path: '/resume',
    handler: getResumeHandler,
  },
  {
    method: 'get',
    path: '/private',
    handler: getPrivateHandler,
    authenticate: true,
  },
  {
    method: 'get',
    path: '/courses',
    handler: getCoursesHandler,
    authenticate: true,
    role: 'engineer',
  },
  {
    method: 'get',
    path: '/admin',
    handler: getAdminHandler,
    authenticate: true,
    role: 'admin',
  },
];


export default routes;
