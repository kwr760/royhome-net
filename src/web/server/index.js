// @flow

import path from 'path';
import express from 'express';
import cors from 'cors';
import helmet from 'helmet';
import compression from 'compression';
import bodyParser from 'body-parser';
import cookieParser from 'cookie-parser';
import httpContext from 'express-http-context';

import env from '@config';

import redirectInsecure from '../../common/server/middleware/redirect-insecure';
import handleError from '../../common/server/middleware/handle-error';
import notFound from '../../common/server/middleware/not-found';
import renderReact from './rendering/render-react';
import startHttpServer from '../../common/server/middleware/start-http';
import startHttpsServer from '../../common/server/middleware/start-https';

const publicDir = path.resolve(env.root);

const app = express<Request, Response>();

app.set('json spaces', 2);
app.enable('etag');
app.enable('query parser');

if (env.server.startHttp) {
  app.use(redirectInsecure);
}

app.use(cors());
app.use(helmet());
app.use(compression());
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({
  extended: false,
}));
app.use(cookieParser());
app.use(httpContext.middleware);

app.use('/', express.static(publicDir));
app.get('/*', renderReact);

app.use(handleError);
app.use(notFound);

startHttpsServer(app, env.server.port);

if (env.server.startHttp) {
  startHttpServer(app, 80);
}

export default app;
