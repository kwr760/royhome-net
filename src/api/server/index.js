// @flow

import express from 'express';
import cors from 'cors';
import helmet from 'helmet';
import compression from 'compression';
import bodyParser from 'body-parser';
import cookieParser from 'cookie-parser';
import httpContext from 'express-http-context';

import env from '@config';

import handleError from '@common/server/middleware/handle-error';
import notFound from '@common/server/middleware/not-found';
import startHttpsServer from '@common/server/middleware/start-https';

import generate from './routes/generate';
import routes from './routes';

const app = express<Request, Response>();

app.set('json spaces', 2);
app.enable('etag');
app.enable('query parser');

app.use(cors());
app.use(helmet());
app.use(compression());
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({
  extended: false,
}));
app.use(cookieParser());
app.use(httpContext.middleware);

app.use('/api', generate(routes));

app.use(handleError);
app.use(notFound);

startHttpsServer(app, env.server.apiPort);

export default app;
