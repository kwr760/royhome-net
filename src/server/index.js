import fs from 'fs';
import path from 'path';
import express from 'express';
import http from 'http';
import https from 'https';
import cors from 'cors';
import helmet from 'helmet';
import compression from 'compression';
import bodyParser from 'body-parser';
import cookieParser from 'cookie-parser';
import httpContext from 'express-http-context';

import redirectInsecure from './middleware/redirect-insecure';
import displayMessage from './middleware/display-message';
import handleError from './middleware/handle-error';
import notFound from './middleware/not-found';

import routes from './routes';
import generate from './routes/generate';

import env from '../config';

const publicDir = path.resolve(env.root, './dist/public');

const app = express();

app.set('json spaces', 2);
app.enable('etag');
app.enable('query parser');

if (env.server.https) {
  app.use(redirectInsecure);
}

app.use(cors());
app.use(helmet());
app.use(compression());
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({
  extended: false,
}));
app.use(cookieParser());
app.use(httpContext.middleware);

app.use('/', express.static(publicDir));
app.use('/callback', express.static(publicDir));
app.use('/api', generate(routes));

app.use(handleError);
app.use(notFound);

if (env.server.https) {
  // Assume that the ports are available on real server.
  const privateKey = fs.readFileSync(env.server.key, 'utf8');
  const certificate = fs.readFileSync(env.server.cert, 'utf8');
  const ca = fs.readFileSync(env.server.ca, 'utf8');

  const credentials = {
    key: privateKey,
    cert: certificate,
    ca,
  };

  const server = https.createServer(credentials, app);
  server.listen(443, displayMessage('Secure server is running'));

  const httpServer = http.createServer(app);
  httpServer.listen(80, displayMessage('Insecure server is being redirected'));
} else {
  // On dev server the ports are locked to root, so use high port
  const httpServer = http.createServer(app);
  httpServer.listen(3000, displayMessage('Server is running'));
}
