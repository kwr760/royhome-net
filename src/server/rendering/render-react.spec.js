import { ChunkExtractor } from '@loadable/server';
import renderReact from './render-react';

jest.mock('@loadable/server');

describe('server/rendering/render-react', () => {
  it('should return a html page', async () => {
    // Arrange
    const req = {
      cookies: {},
    };
    const res = {
      send: jest.fn(),
      sendStatus: jest.fn(),
    };
    ChunkExtractor.mockImplementation(() => ({
      requireEntrypoint: jest.fn(() => ({ default: jest.fn() })),
      collectChunks: jest.fn(() => '<div>Chunks</div>'),
      getLinkTags: jest.fn(() => '<div>Links</div>'),
      getStyleTags: jest.fn(() => '<div>Styles</div>'),
      getScriptTags: jest.fn(() => '<div>Scripts</div>'),
    }));

    // Act
    await renderReact(req, res);

    // Assert
    expect(res.send).toMatchSnapshot();
  });
});
