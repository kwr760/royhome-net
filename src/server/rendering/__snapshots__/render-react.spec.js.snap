// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server/rendering/render-react should return a html page 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "<!DOCTYPE html>
<html lang=\\"en\\" class=\\"h-100\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <title>Roy Home</title>
    <base href=\\"/\\" >
    <link rel=\\"shortcut icon\\" href=\\"/dist/web/favicon.ico\\">
    <link href=\\"https://use.fontawesome.com/releases/v5.12.1/css/svg-with-js.css\\" rel=\\"stylesheet\\">
    <div>Links</div>
    <div>Styles</div>
    <script>window.__PRELOADED_STATE__ = {\\"session\\":{\\"authenticated\\":false,\\"expiration\\":-1,\\"isLoading\\":false},\\"user\\":{},\\"resume\\":{\\"activeResume\\":\\"kroy760@gmail.com\\",\\"kroy760@gmail.com\\":{\\"owner\\":{\\"name\\":\\"Kevin Roy\\"},\\"contact\\":{\\"phone\\":\\"(425) 208-1223\\",\\"email\\":\\"kroy760@gmail.com\\",\\"displayPhone\\":false},\\"address\\":{\\"address\\":\\"Issaquah, WA  98027\\"},\\"summary\\":{\\"summary\\":\\"Proactive and results-oriented professional with experience as technical leader on fast-paced and time-critical projects. Aggressive at overcoming challenges by developing robust solutions. Exceptional communicator and practitioner of full-cycle communications with capacity to independently manage complex software projects and support organizational customer service, quality assurance and sales goals.\\"},\\"skills\\":[{\\"id\\":1,\\"position\\":1,\\"name\\":\\"Languages\\",\\"items\\":[{\\"id\\":11,\\"position\\":1,\\"name\\":\\"PHP\\"},{\\"id\\":10,\\"position\\":2,\\"name\\":\\"HTML\\"},{\\"id\\":9,\\"position\\":3,\\"name\\":\\"JavaScript\\"},{\\"id\\":8,\\"position\\":4,\\"name\\":\\"CSS\\"},{\\"id\\":7,\\"position\\":5,\\"name\\":\\"C\\"},{\\"id\\":6,\\"position\\":6,\\"name\\":\\"C++\\"},{\\"id\\":5,\\"position\\":7,\\"name\\":\\"Java\\"},{\\"id\\":4,\\"position\\":8,\\"name\\":\\"Perl\\"},{\\"id\\":3,\\"position\\":9,\\"name\\":\\"bash\\"},{\\"id\\":2,\\"position\\":10,\\"name\\":\\"XML\\"},{\\"id\\":1,\\"position\\":11,\\"name\\":\\"json\\"}]},{\\"id\\":2,\\"position\\":2,\\"name\\":\\"Frameworks and Toolkits\\",\\"items\\":[{\\"id\\":17,\\"position\\":1,\\"name\\":\\"jquery\\"},{\\"id\\":16,\\"position\\":2,\\"name\\":\\"ajax\\"},{\\"id\\":15,\\"position\\":3,\\"name\\":\\"PHPUnit\\"},{\\"id\\":14,\\"position\\":4,\\"name\\":\\"PL/SQL\\"},{\\"id\\":13,\\"position\\":5,\\"name\\":\\"node.js\\"},{\\"id\\":12,\\"position\\":6,\\"name\\":\\"bootstrap\\"}]},{\\"id\\":3,\\"position\\":3,\\"name\\":\\"Databases\\",\\"items\\":[{\\"id\\":20,\\"position\\":1,\\"name\\":\\"MySQL\\"},{\\"id\\":19,\\"position\\":2,\\"name\\":\\"Oracle\\"},{\\"id\\":18,\\"position\\":3,\\"name\\":\\"SQL Server\\"}]},{\\"id\\":4,\\"position\\":4,\\"name\\":\\"Tools\\",\\"items\\":[{\\"id\\":32,\\"position\\":1,\\"name\\":\\"git\\"},{\\"id\\":31,\\"position\\":2,\\"name\\":\\"Jira\\"},{\\"id\\":30,\\"position\\":3,\\"name\\":\\"phanthomjs\\"},{\\"id\\":29,\\"position\\":4,\\"name\\":\\"Selenium\\"},{\\"id\\":28,\\"position\\":5,\\"name\\":\\"maven\\"},{\\"id\\":27,\\"position\\":6,\\"name\\":\\"aws\\"},{\\"id\\":26,\\"position\\":7,\\"name\\":\\"IntelliJ\\"},{\\"id\\":25,\\"position\\":8,\\"name\\":\\"Eclipse\\"},{\\"id\\":24,\\"position\\":9,\\"name\\":\\"vim\\"},{\\"id\\":23,\\"position\\":10,\\"name\\":\\"clearcase\\"},{\\"id\\":22,\\"position\\":11,\\"name\\":\\"docker\\"},{\\"id\\":21,\\"position\\":12,\\"name\\":\\"splunk\\"}]},{\\"id\\":5,\\"position\\":5,\\"name\\":\\"Platforms\\",\\"items\\":[{\\"id\\":35,\\"position\\":1,\\"name\\":\\"Apache\\"},{\\"id\\":34,\\"position\\":2,\\"name\\":\\"JBoss\\"},{\\"id\\":33,\\"position\\":3,\\"name\\":\\"LAMP\\"}]},{\\"id\\":6,\\"position\\":6,\\"name\\":\\"Methodologies\\",\\"items\\":[{\\"id\\":39,\\"position\\":1,\\"name\\":\\"SDLC\\"},{\\"id\\":38,\\"position\\":2,\\"name\\":\\"Agile\\"},{\\"id\\":37,\\"position\\":3,\\"name\\":\\"Scrum\\"},{\\"id\\":36,\\"position\\":4,\\"name\\":\\"Waterfall\\"}]},{\\"id\\":7,\\"position\\":7,\\"name\\":\\"Operating Systems\\",\\"items\\":[{\\"id\\":48,\\"position\\":1,\\"name\\":\\"Ubuntu\\"},{\\"id\\":47,\\"position\\":2,\\"name\\":\\"Windows\\"},{\\"id\\":46,\\"position\\":3,\\"name\\":\\"Red Hat\\"},{\\"id\\":45,\\"position\\":4,\\"name\\":\\"HP-UX\\"},{\\"id\\":44,\\"position\\":5,\\"name\\":\\"MAC/OS\\"},{\\"id\\":43,\\"position\\":6,\\"name\\":\\"Solaris\\"},{\\"id\\":42,\\"position\\":7,\\"name\\":\\"SCO\\"},{\\"id\\":41,\\"position\\":8,\\"name\\":\\"AIX\\"},{\\"id\\":40,\\"position\\":9,\\"name\\":\\"AS/400\\"}]}],\\"experience\\":[{\\"id\\":1,\\"position\\":1,\\"title\\":\\"Technical Lead\\",\\"company\\":\\"Sterling Talent Solutions\\",\\"startDate\\":\\"May       2012\\",\\"description\\":[{\\"id\\":19,\\"position\\":1,\\"type\\":\\"text\\",\\"item\\":\\"A key contributor on developing a product in a small aggressive company to be acquired by a large industry leader. We provided quality background screening and onboarding solutions. Worked in major areas of the company and developed multiple large features individually and as a leader in the team.\\"},{\\"id\\":18,\\"position\\":2,\\"type\\":\\"text\\",\\"item\\":\\"I worked on two major aspects of the company's product. The first is a highly configurable form based data collection framework. The second being the position and processing of the various searches and the compliance around them.\\"}],\\"bullets\\":[{\\"id\\":17,\\"position\\":3,\\"type\\":\\"bullet\\",\\"item\\":\\"Fair Chance: Implemented the current states requirements for a process similar to Individualized Assessment including filling out state provided forms.\\"},{\\"id\\":16,\\"position\\":4,\\"type\\":\\"bullet\\",\\"item\\":\\"Individualized Assessment: Implemented the ability for the customer to request and process a candidate's response to an adverse action being preformed on a candidate.\\"},{\\"id\\":15,\\"position\\":5,\\"type\\":\\"bullet\\",\\"item\\":\\"eDispute: Implemented the ability for candidate to review and dispute the reports or searches preformed on them.\\"},{\\"id\\":14,\\"position\\":6,\\"type\\":\\"bullet\\",\\"item\\":\\"Form Review: Implemented the customers ability to review the customized forms, including sending the form back to the candidate for modification.\\"},{\\"id\\":13,\\"position\\":7,\\"type\\":\\"bullet\\",\\"item\\":\\"Esignature: Developed the signing experience to provide confirmation that the user is certifying their actions. Including the generate of a PDF from HTML.\\"},{\\"id\\":12,\\"position\\":8,\\"type\\":\\"bullet\\",\\"item\\":\\"Candidate Model Export: Provided a feature to enable customer to export the data about their candidates.\\"},{\\"id\\":11,\\"position\\":9,\\"type\\":\\"bullet\\",\\"item\\":\\"Professional Services Billing System: Enable another team to easily bill customer based on their individual contracts.\\"}],\\"techs\\":[{\\"id\\":10,\\"position\\":10,\\"type\\":\\"tech\\",\\"item\\":\\"Ubuntu\\"},{\\"id\\":9,\\"position\\":11,\\"type\\":\\"tech\\",\\"item\\":\\"Apache\\"},{\\"id\\":8,\\"position\\":12,\\"type\\":\\"tech\\",\\"item\\":\\"MySQL\\"},{\\"id\\":7,\\"position\\":13,\\"type\\":\\"tech\\",\\"item\\":\\"Php\\"},{\\"id\\":6,\\"position\\":14,\\"type\\":\\"tech\\",\\"item\\":\\"Html\\"},{\\"id\\":5,\\"position\\":15,\\"type\\":\\"tech\\",\\"item\\":\\"JavaScript\\"},{\\"id\\":4,\\"position\\":16,\\"type\\":\\"tech\\",\\"item\\":\\"Css\\"},{\\"id\\":3,\\"position\\":17,\\"type\\":\\"tech\\",\\"item\\":\\"phantomJs\\"},{\\"id\\":2,\\"position\\":18,\\"type\\":\\"tech\\",\\"item\\":\\"IntelliJ\\"},{\\"id\\":1,\\"position\\":19,\\"type\\":\\"tech\\",\\"item\\":\\"PhpUnit\\"}]},{\\"id\\":2,\\"position\\":2,\\"title\\":\\"Senior Software Engineer\\",\\"company\\":\\"Applied Discovery, Inc.\\",\\"startDate\\":\\"October   2010\\",\\"endDate\\":\\"April     2012\\",\\"description\\":[{\\"id\\":29,\\"position\\":1,\\"type\\":\\"text\\",\\"item\\":\\"Worked on team to design the next generation e-Discovery processing line using open source technologies.\\"}],\\"bullets\\":[{\\"id\\":28,\\"position\\":2,\\"type\\":\\"bullet\\",\\"item\\":\\"Researched and evaluated various technologies: Parallel Processing Framework and grid (Globus/Condor/JPPF), workflow technologies (jPBM), JBoss, RESTful, Hibernate, Boost, Oracle’s OutsideIn, Google’s Guice, Talend and others.\\"},{\\"id\\":27,\\"position\\":3,\\"type\\":\\"bullet\\",\\"item\\":\\"Implemented Straw man of new system written in Java using a RESTful API.\\"},{\\"id\\":26,\\"position\\":4,\\"type\\":\\"bullet\\",\\"item\\":\\"Implemented an Object Oriented Application (C++) to utilize Oracle’s OutsideIn Technology to parse and load Microsoft Outlook pst file.\\"}],\\"techs\\":[{\\"id\\":25,\\"position\\":5,\\"type\\":\\"tech\\",\\"item\\":\\"Java\\"},{\\"id\\":24,\\"position\\":6,\\"type\\":\\"tech\\",\\"item\\":\\"C\\"},{\\"id\\":23,\\"position\\":7,\\"type\\":\\"tech\\",\\"item\\":\\"C++\\"},{\\"id\\":22,\\"position\\":8,\\"type\\":\\"tech\\",\\"item\\":\\"Windows\\"},{\\"id\\":21,\\"position\\":9,\\"type\\":\\"tech\\",\\"item\\":\\"Unix\\"},{\\"id\\":20,\\"position\\":10,\\"type\\":\\"tech\\",\\"item\\":\\"SQL Server\\"}]},{\\"id\\":3,\\"position\\":3,\\"title\\":\\"Software Engineer/Technical Lead\\",\\"company\\":\\"SofTech, Inc.\\",\\"startDate\\":\\"March     2001\\",\\"endDate\\":\\"October   2010\\",\\"description\\":[{\\"id\\":43,\\"position\\":1,\\"type\\":\\"text\\",\\"item\\":\\"Worked on a product lifecycle management product to provide new features, maintenance and customer support for medium-sized businesses. Managed Pro/C coding, data, large configuration loading, APIs, licensing, installations and research third-party product technical issues.\\"}],\\"bullets\\":[{\\"id\\":42,\\"position\\":2,\\"type\\":\\"bullet\\",\\"item\\":\\"Led technical effort to internationalize products into local languages that allowed sales in foreign markets.\\"},{\\"id\\":41,\\"position\\":3,\\"type\\":\\"bullet\\",\\"item\\":\\"Designed and implemented revision editing tool that enabled administrators to define multiple revision sequences and apply them to classes, which increased value of company’s product line.\\"},{\\"id\\":40,\\"position\\":4,\\"type\\":\\"bullet\\",\\"item\\":\\"Built feature that pushed files into replication servers to increase performance over the WANs.\\"},{\\"id\\":39,\\"position\\":5,\\"type\\":\\"bullet\\",\\"item\\":\\"Transited to server-based report generation using libxml2 and libxslt technology to improve performance.\\"},{\\"id\\":38,\\"position\\":6,\\"type\\":\\"bullet\\",\\"item\\":\\"Improved querying ability and performance by implementing a list like clause.\\"},{\\"id\\":37,\\"position\\":7,\\"type\\":\\"bullet\\",\\"item\\":\\"Redesigned aspects of the product to enable developers to use QT technology that minimized maintenance costs and enabled code reuse.\\"},{\\"id\\":36,\\"position\\":8,\\"type\\":\\"bullet\\",\\"item\\":\\"Cross over to new code layers to resolve problems; demonstrated extra effort in fixing root causes, keeping code simple, removing unused code and using build tools that increase productivity.\\"},{\\"id\\":35,\\"position\\":9,\\"type\\":\\"bullet\\",\\"item\\":\\"Took initiative to rewrite header file architecture, which saved up to two hours of compile time.\\"}],\\"techs\\":[{\\"id\\":34,\\"position\\":10,\\"type\\":\\"tech\\",\\"item\\":\\"C\\"},{\\"id\\":33,\\"position\\":11,\\"type\\":\\"tech\\",\\"item\\":\\"C++\\"},{\\"id\\":32,\\"position\\":12,\\"type\\":\\"tech\\",\\"item\\":\\"Unix\\"},{\\"id\\":31,\\"position\\":13,\\"type\\":\\"tech\\",\\"item\\":\\"SQL\\"},{\\"id\\":30,\\"position\\":14,\\"type\\":\\"tech\\",\\"item\\":\\"Perl\\"}]},{\\"id\\":4,\\"position\\":4,\\"title\\":\\"Contractor/Software Engineer\\",\\"company\\":\\"OrderTrust\\",\\"startDate\\":\\"February  2000\\",\\"endDate\\":\\"March     2001\\",\\"description\\":[{\\"id\\":51,\\"position\\":1,\\"type\\":\\"text\\",\\"item\\":\\"Worked on main payment application in UNIX using C++ and Informix for provider of marketing and order management services for online retailers and catalogs. Developed prototype value-added payment program in J2EE and loyalty-based system of electronic coupons. Cleaned code and automated processes.\\"}],\\"bullets\\":[{\\"id\\":50,\\"position\\":2,\\"type\\":\\"bullet\\",\\"item\\":\\"Simplified code maintenance of entire system by moving legacy code into standardized build environment.\\"},{\\"id\\":49,\\"position\\":3,\\"type\\":\\"bullet\\",\\"item\\":\\"Automated testing Dining A La Card system with new systems using Perl script that improved accuracy and reduced time needed to test systems and removed element of human error.\\"}],\\"techs\\":[{\\"id\\":48,\\"position\\":4,\\"type\\":\\"tech\\",\\"item\\":\\"C++\\"},{\\"id\\":47,\\"position\\":5,\\"type\\":\\"tech\\",\\"item\\":\\"Unix\\"},{\\"id\\":46,\\"position\\":6,\\"type\\":\\"tech\\",\\"item\\":\\"SQL\\"},{\\"id\\":45,\\"position\\":7,\\"type\\":\\"tech\\",\\"item\\":\\"Java\\"},{\\"id\\":44,\\"position\\":8,\\"type\\":\\"tech\\",\\"item\\":\\"Perl\\"}]},{\\"id\\":5,\\"position\\":5,\\"title\\":\\"Software Engineer\\",\\"company\\":\\"PSW Technology\\",\\"startDate\\":\\"April     1999\\",\\"endDate\\":\\"February  2000\\",\\"description\\":[{\\"id\\":55,\\"position\\":1,\\"type\\":\\"text\\",\\"item\\":\\"Worked for consulting company for the IBM/Lotus Notes with 10 employees to port Lotus Notes and toolkits to IBM AS/400 system and to correct bugs. Developed and convinced engineers to buy into solutions to problems.\\"}],\\"bullets\\":[{\\"id\\":54,\\"position\\":2,\\"type\\":\\"bullet\\",\\"item\\":\\"Built portable code using a cross-compiler and corrected bugs using native AS/400 debugger.\\"}],\\"techs\\":[{\\"id\\":53,\\"position\\":3,\\"type\\":\\"tech\\",\\"item\\":\\"C\\"},{\\"id\\":52,\\"position\\":4,\\"type\\":\\"tech\\",\\"item\\":\\"AS/400\\"}]},{\\"id\\":6,\\"position\\":6,\\"title\\":\\"Contractor\\",\\"company\\":\\"Celestica, Inc.\\",\\"startDate\\":\\"May       1997\\",\\"endDate\\":\\"April     1999\\",\\"description\\":[{\\"id\\":61,\\"position\\":1,\\"type\\":\\"text\\",\\"item\\":\\"Developed, repaired and enhanced programs and processes for electronics manufacturer using in-house client/server application system to manage flow of process lines. Developed and debugged client/server applications in C, Informix Embedded SQL and Korn Shell and created new modules for integration with X Windows program and ESQL.\\"}],\\"bullets\\":[{\\"id\\":60,\\"position\\":2,\\"type\\":\\"bullet\\",\\"item\\":\\"Redesigned process of changing bill-of-materials workstation structures that avoided deadlocks.\\"},{\\"id\\":59,\\"position\\":3,\\"type\\":\\"bullet\\",\\"item\\":\\"Fixed key user interface, enabling users to step through process of building workstations and PCs.\\"},{\\"id\\":58,\\"position\\":4,\\"type\\":\\"bullet\\",\\"item\\":\\"Determined report needs of users that included finding lost workstations and bottlenecks and implemented intranet-accessible reports using C, Korn Shell, HTML and Sapphire CGI generator.\\"}],\\"techs\\":[{\\"id\\":57,\\"position\\":5,\\"type\\":\\"tech\\",\\"item\\":\\"C/C++\\"},{\\"id\\":56,\\"position\\":6,\\"type\\":\\"tech\\",\\"item\\":\\"HP/UX\\"}]},{\\"id\\":7,\\"position\\":7,\\"title\\":\\"Project Leader/Software Engineer\\",\\"company\\":\\"Boston Technology, Inc.\\",\\"startDate\\":\\"December  1995\\",\\"endDate\\":\\"May       1997\\",\\"description\\":[{\\"id\\":65,\\"position\\":1,\\"type\\":\\"text\\",\\"item\\":\\"Worked in the leading communications company that implemented voicemail systems for large telecommunications companies to lead feature development and bug fixing projects. Including a traffic statistic feature for voicemail application and fixed bugs for various international projects written in C/C++ and fixed bugs in billing systems.\\"}],\\"bullets\\":[],\\"techs\\":[{\\"id\\":64,\\"position\\":2,\\"type\\":\\"tech\\",\\"item\\":\\"C/C++\\"},{\\"id\\":63,\\"position\\":3,\\"type\\":\\"tech\\",\\"item\\":\\"SCO Unix\\"},{\\"id\\":62,\\"position\\":4,\\"type\\":\\"tech\\",\\"item\\":\\"SQL\\"}]},{\\"id\\":8,\\"position\\":8,\\"title\\":\\"Software Engineer\\",\\"company\\":\\"ESSENSE Systems, Inc.\\",\\"startDate\\":\\"August    1994\\",\\"endDate\\":\\"December  1995\\",\\"description\\":[{\\"id\\":66,\\"position\\":1,\\"type\\":\\"text\\",\\"item\\":\\"Developed aspects of a data-driven GUI that empowered employees to manage their own human resource records.\\"}],\\"bullets\\":[],\\"techs\\":[]}],\\"education\\":[{\\"degree\\":\\"Master of Science in Computer Science\\",\\"school\\":\\"University of Massachusetts at Lowell\\",\\"graduationDate\\":\\"February  2001\\"},{\\"degree\\":\\"Bachelor of Science in Computer Science\\",\\"school\\":\\"Worcester Polytechnic Institute\\",\\"graduationDate\\":\\"May       1994\\"}]}}}</script>
  </head>
  <body class=\\"h-100 backdrop\\">
    <div id=\\"main\\" class=\\"h-100\\">&lt;div&gt;Chunks&lt;/div&gt;</div>
    <div>Scripts</div>
  </body>
</html>
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
